src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -488,11 +488,12 @@ protected void writeToNBT(@NotNull final NBTTagCompound compound)
 
         //  Achievements
         @NotNull final NBTTagList achievementsTagList = new NBTTagList();
-        for (@NotNull final Achievement achievement : this.colonyAchievements)
+        for (@NotNull final Advancement achievement : this.colonyAchievements)
         {
             @NotNull final NBTTagCompound achievementCompound = new NBTTagCompound();
-            achievementCompound.setString(TAG_ACHIEVEMENT, achievement.statId);
-            achievementsTagList.appendTag(achievementCompound);
+            //todo deserialization
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)